Add eus channel info (#27594)
openshift/openshift-docs: modules/understanding-upgrade-channels.adoc

* `candidate-{product-version}`
* `fast-{product-version}`
* `stable-{product-version}`

> * `eus-4.6` (only available when running 4.6)

Reviewer: This should say, when running an EUS version, as we will have more EUS versions than just 4.6

Author: This is accurate as written, you will not see the eus-4.6 channel unless you're already running 4.6 versions and we don't have to worry about other eus channels until next November, we can deal with addition of subsequent EUS channels then.

> * `eus-4.6` (only available when running 4.6)

Reviewer 2: Is this channel available now? I did not see it from a v4.6.6 cluster. But following date in [1] says it's available on October 27, 2020.
```
Version	General availability	Full support ends	Maintenance support ends
4.6 EUS	October 27, 2020	Release of 4.7 + 1 month	Release of 4.7 + 14 months
```
[1] https://access.redhat.com/support/policy/updates/openshift#ocp4_phases

Author: Not yet, we'll land these documentation changes and the new channel at the same time.

Reviewer 2: ok, then lgtm.

Reviewer 3: > Not yet, we'll land these documentation changes and the new channel at roughly the same time as we ship 4.6.7, that's the plan at least.

@sdodson, do you want me merge this change after the 4.6.7. release notes go out?

Author: I'm not too concerned about the ordering as long as it's roughly around the same time.

Reviewer 4: @sdodson the release notes for 4.6.8 (including the callout for eus-4.6) were published this morning and are now available: https://docs.openshift.com/container-platform/4.6/release_notes/ocp-4-6-release-notes.html#ocp-4-6-8-eus-channel. Should we proceed in merging this now?

Author: yes please! the channel is now live as well.

You can use the `stable-{product-version}` channel to upgrade from a previous minor version of
{product-title}.

[discrete]
== eus-{product-version} channel

In addition to the stable channel, certain minor versions of {product-title} offer an
Extended Update Support[https://access.redhat.com/support/policy/updates/openshift#ocp4_phases] (EUS).
These EUS versions extend the maintenance phase for customers with Premium Subscriptions

> to 14 months. {product-title} {product-version} is a minor which includes EUS.

Reviewer 3: ```suggestion
to 14 months. {product-title} {product-version} is a minor version that includes EUS.
```
If not all versions are going to be EUS, I also suggest not using the {product-version} attribute and manually specifying the version instead.

> While there's no difference between stable-{product-version} and eus-{product-version}

Reviewer 3: ```suggestion
Although there is no difference between stable-{product-version} and eus-{product-version}
```

> channels until {product-version} transitions to EUS phase customers may switch

Reviewer 3: ```suggestion
channels until {product-version} transitions to EUS phase, you can switch
```

> channels until {product-version} transitions to EUS phase, you can switch

Reviewer 5: "to EUS phase" -> "to the EUS phase"?

Author: how about simply "to EUS, you may switch..." ?

Author: nevermind, I'll go with "to the EUS phase..."

to the EUS channel as soon as it becomes available. When {product-title} {product-version}

> transitions to the EUS phase of its lifecycle the stable-{product-version} will

Reviewer 3: ```suggestion
transitions to the EUS phase of its lifecycle, the stable-{product-version} channel will
```

> no longer receive updates. Once a cluster upgrades to a version that's exclusive

Reviewer 3: ```suggestion
no longer receive updates. After you upgrade a to a version that is exclusive
```

> to the EUS channel that cluster will no longer be eligible for minor version upgrades

Reviewer 3: ```suggestion
to the EUS channel, that cluster is no longer be eligible for minor version upgrades
```

> to the EUS channel, that cluster is no longer be eligible for minor version upgrades

Reviewer 5: "is no longer be eligible" -> "will no longer be eligible" or "is no longer eligible"

until upgrades to the next EUS version become available. Currently, upgrades between

> EUS versions will entail a serial set of upgrades, ie: 4.6 to 4.7 to 4.8 to 4.9 to 4.10.

Reviewer 3: ```suggestion
EUS versions requires a serial set of version upgrades, such as from 4.6 to 4.7 to 4.8 to 4.9 to 4.10.
```

> EUS versions requires a serial set of version upgrades, such as from 4.6 to 4.7 to 4.8 to 4.9 to 4.10.

Reviewer 5: Is this an important aspect of the EUS channel?  Can we say something like:

> For at least the final six months (or whatever) of the EUS phase, an update pathway to the next EUS version will be available.

without going into detail about what we expect that update pathway will look like?

Author: I feel like we owe them some level of expectation setting, right now it's very likely this will involve four serialized upgrades. Without work getting put into planning to improve that I'd rather be upfront about the worst case.

Additionally, you must upgrade to {product-title} {product-version} before it

> transitions to EUS phase, ie: you cannot upgrade from 4.5 to 4.6 after 4.6 transitions

Reviewer 3: ```suggestion
transitions to EUS phase. For example, you cannot upgrade from 4.5 to 4.6 after 4.6 transitions
```
I'm a little confused about this last paragraph. I feel like there's a bit of context missing, maybe an assumption about the type of channel that you're in before this switchover happens? What happens if you don't upgrade from 4.5 quickly enough? Are you stuck on 4.5 forever? Will the graph let you move from 4.5 to 4.7?

Author: The context is that we'll stop testing upgrades from 4.5 to the 4.6 EUS versions. While nothing would prevent them from upgrading from 4.5 to 4.6 on stable channel  then switching to EUS 4.6 channel I don't think we want to encourage that sort of extreme lag. If they wait that long not only has 4.5 gone EOL but 4.6, with the exception of EUS, has too.

This is a bit of an open question that I'm working through with PM but I do assume they're going to be in support of this limitation.

> to EUS status.

Reviewer 5: Is this an important aspect of the EUS phase?  By the time 4.6 hits end-of-maintenance (and the beginning of EUS), 4.5 will have been EOL for a while, so those folks are already out of support.  Say 4.5.99 is the last 4.5.z, and 4.6.90 is the last end-of-maintenance 4.6.z.  Folks who are on 4.5.99 when we enter EUS with 4.6.91 can still go 4.5.99 -> 4.6.90 -> 4.6.91, without going more stale than they were when they were running 4.5.99.  The fact that we are not currently planning to add direct 4.5.99 -> 4.6.91 edges does not seem important enough to be worth calling out in the channel docs.

Author: I just simplified it to say that you have to be running a supported version of 4.6 before switching to eus-4.6. PTAL

You can use the `stable-{product-version}` channel to upgrade from a previous minor version of
{product-title}.

[discrete]
== eus-4.6 channel

In addition to the stable channel, certain minor versions of {product-title} offer an
Extended Update Support[https://access.redhat.com/support/policy/updates/openshift#ocp4_phases] (EUS).
These EUS versions extend the maintenance phase for customers with Premium Subscriptions
to 14 months. {product-title} 4.6 is currently the only minor version with EUS.

Although there is no difference between stable-4.6 and eus-4.6
channels until {product-title} 4.6 transitions to EUS the phase, you can switch
to the EUS channel as soon as it becomes available. When {product-title} 4.6
transitions to the EUS phase of its lifecycle the stable-4.6 channel will
no longer receive subsequent z-stream updates. After you upgrade a to a version that

> is exclusive to the EUS channel, that cluster will no longer be eligible for minor

Reviewer 2: "upgrade a to" should be "upgrade a cluster to"?

Author: thanks, fixed that

> version upgrades until upgrades to the next EUS version become available. Currently,

Author: Thanks, added something to that effect.

planned EUS version is to be 4.10 and the upgrade to that version will require a
serial set of version upgrades, such as from 4.6 to 4.7 to 4.8 to 4.9 to 4.10.

> 

Reviewer 2: Does it mean, only when 4.10 EUS is available, we have to upgrade a 4.6 EUS cluster to v4.7 first, and then 4.8-4.9-4.10, but not upgrade it to 4.10 directly from the EUS channel? And before that, no v4.7 will be available for a 4.6 EUS channel?

Author: Yes, when upgrades from 4.6 EUS to 4.10 become available you'd switch to the eus-4.10 channel and then repeatedly upgrade from 4.6 to 4.7 to 4.8 to 4.9 to 4.10. There's no planned direct 4.6 to 4.10 upgrade at this time.

Reviewer 2: get it, thanks.
